app/src/main/java/com/pomonyang/mohanyang/notification/MnFirebaseMessagingService.kt

```diff
@@ -0,0 +1,64 @@
+package com.pomonyang.mohanyang.notification
+
+import android.Manifest
+import android.content.pm.PackageManager
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationManagerCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.pomonyang.mohanyang.data.repository.push.PushAlarmRepository
+import com.pomonyang.mohanyang.notification.util.defaultNotification
+import dagger.hilt.android.AndroidEntryPoint
+import java.util.*
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+internal class MnFirebaseMessagingService : FirebaseMessagingService() {
+    private val scope = CoroutineScope(Dispatchers.IO + SupervisorJob())
+
+    @Inject
+    lateinit var pushAlarmRepository: PushAlarmRepository
+
+    private val isPushEnabled = true // TODO 사용자 정보 가져오기
+
+    override fun onNewToken(token: String) {
+        if (isPushEnabled) {
+            scope.launch {
+                pushAlarmRepository.apply {
+                    saveFcmToken(token)
+                    registerPushToken(token)
+                }
+            }
+        }
+    }
+
+    override fun onMessageReceived(remoteMessage: RemoteMessage) {
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) {
+            return
+        }
+
+        if (!isPushEnabled) return
+
+        val notification = remoteMessage.notification ?: return
+
+        val builder = applicationContext.defaultNotification()
+        builder.apply {
+            setContentTitle(notification.title)
+            setContentText(notification.body)
+        }
```

Comment:
[무반응.webm](https://github.com/user-attachments/assets/6d725b3b-7009-4b86-940a-89fc2436921d)

동작성을 보고 있는데 위처럼 앱이 Foreground에서는 `onMessageReceived`를 타서 우리가 만든 Noti를 띄워주는거다보니 설정들을 안해주면 알림 클릭했을 때 무반응이던데 아래처럼 intent 설정을 해주는건 어떨까?

```suggestion
         val pendingIntent = PendingIntent.getActivity(
            applicationContext,
            UUID.randomUUID().hashCode(),
            Intent(applicationContext, MainActivity::class.java),
            PendingIntent.FLAG_ONE_SHOT or PendingIntent.FLAG_IMMUTABLE //일회용 펜딩 인텐트
        )

        val builder = applicationContext.defaultNotification()
        builder.apply {
            setContentTitle(notification.title)
            setContentText(notification.body)
            setContentIntent(pendingIntent)
            setAutoCancel(true)
        }
```

Reply:
디테일 아주 좋아요 바로 이렇게 수정해서 가면 될거 같아 bb

```kotlin
  val pendingIntent = PendingIntent.getActivity(
            applicationContext,
            UUID.randomUUID().hashCode(),
            Intent(applicationContext, MainActivity::class.java),
            PendingIntent.FLAG_ONE_SHOT or PendingIntent.FLAG_IMMUTABLE // 일회용 펜딩 인텐트
        )

        val builder = applicationContext.defaultNotification(pendingIntent)
```